### ISSUE TITLE
Spectrogram cannot be destroyed

### ISSUE THREAD

**Issue by jamesshannon** [katspaugh/wavesurfer.js#1571]
### Code needed to reproduce the issue:

1. Create wavesurfer, with the spectrogram plugin
2. Optionally, load a sound
3. Destroy spectogram plugin (e.g., destroyAllPlugins())
4. Load a sound

An error occurs. I can't remember it exactly, but it's due to an instance not being available when called by the event code. The spectogram plugin [listens to the redraw event](https://github.com/katspaugh/wavesurfer.js/blob/master/src/plugin/spectrogram.js#L321) but [only detaches from the onReady event](https://github.com/katspaugh/wavesurfer.js/blob/master/src/plugin/spectrogram.js#L336).( Looks like it also sets an HTML event, though I wasn't getting exceptions for that.)

**Comment by thijstriemstra** [comment]
I can't reproduce this on https://wavesurfer-js.org/plugins/spectrogram.html for example.

Running `wavesurfer.destroyAllPlugins()` in the console simply removes plugin as expected.

We'll need the exact error with code to reproduce @jamesshannon

**Comment by jamesshannon** [comment]
Did you then try to load and play another sound? This is what I'm doing:

```
var wavesurfer = WaveSurfer.create({
      container: '#waveform',
      plugins: [
        WaveSurfer.spectrogram.create({
            wavesurfer: wavesurfer,
            container: "#spectrogram"
        })
      ]
    });

   wavesurfer.on('ready', function () {
      wavesurfer.play()
    });

    wavesurfer.load('/recording/20190220/click_recording_2018-12-10T210542.wav');

    // wait for sound to play
    wavesurfer.destroyAllPlugins()

     wavesurfer.load('/recording/20190220/click_recording_2018-12-10T210837.wav');
```
I think get this exception:
```
Uncaught TypeError: Cannot read property 'backend' of null
    at SpectrogramPlugin.getFrequencies (spectrogram.js:465)
    at SpectrogramPlugin.render (spectrogram.js:422)
    at spectrogram.js:321
    at observer.js:116
    at Array.forEach (<anonymous>)
    at WaveSurfer.fireEvent (observer.js:115)
    at WaveSurfer.drawBuffer (wavesurfer.js:1100)
    at WaveSurfer.loadDecodedBuffer (wavesurfer.js:1152)
    at wavesurfer.js:1138
    at wavesurfer.js:1334
```

**Comment by thijstriemstra** [comment]
> Did you then try to load and play another sound?

Your ticket doesn't include an expectation. So first of all, what do you expect to happen?

**Comment by jamesshannon** [comment]
For this issue my expectation is that I don't get an exception, and that wavesurfer does things... like play a sound. However, since there's an exception, wavesurfer doesn't do things, like playing sounds.

**Comment by thijstriemstra** [comment]
after you destroy a plugin, you expect wavesurfer still to function properly when loading a sound?

**Comment by jamesshannon** [comment]
Yes. Is that not reasonable? :)

Firstly, the reason I took this approach is the "leak" that I detailed in #1572. My thought was that I could destroy and re-add the spectrogram every time I load a sound in order to prevent the leaks. Since I've fixed that locally then this use case is less important to me.

However, I would still expect that I can destroy a plugin after runtime and that the other functionality would still work. The docs don't say otherwise. And it's not clear to me why `destroy()`ing specific plugins would even be exposed (at least externally) if you're not able to continue using wavesurfer afterwards.
